AIDevGallery/Pages/Models/ModelPage.xaml.cs

```diff
@@ -155,6 +163,74 @@ private IEnumerable<ModelDetails> GetAllSampleDetails()
         }
     }
 
+    private void BuildAIToolkitButton()
+    {
+        bool isAiToolkitActionAvailable = false;
+        Dictionary<AIToolkitAction, MenuFlyoutSubItem> actionSubmenus = new();
+
+        foreach(ModelDetails modelDetails in models)
+        {
+            foreach(AIToolkitAction action in modelDetails.AIToolkitActions!)
+            {
+                if(!AIToolkitHelper.ValidateAction(modelDetails, action))
+                {
+                    continue;
+                }
+
+                MenuFlyoutSubItem? actionFlyoutItem;
+                if (!actionSubmenus.TryGetValue(action, out actionFlyoutItem))
+                {
+                    actionFlyoutItem = new MenuFlyoutSubItem()
+                    {
+                        Text = AIToolkitHelper.AIToolkitActionInfos[action].DisplayName
+                    };
+                    actionSubmenus.Add(action, actionFlyoutItem);
+                    AIToolkitFlyout.Items.Add(actionFlyoutItem);
+                }
+
+                isAiToolkitActionAvailable = true;
+                MenuFlyoutItem modelFlyoutItem = new MenuFlyoutItem()
+                {
+                    Tag = (action, modelDetails),
+                    Text = modelDetails.Name,
+                };
+
+                modelFlyoutItem.Click += ToolkitActionFlyoutItem_Click;
+                actionFlyoutItem.Items.Add(modelFlyoutItem);
+            }
+        }
+
+        AIToolkitDropdown.Visibility = isAiToolkitActionAvailable ? Visibility.Visible : Visibility.Collapsed;
+    }
+
+    private void ToolkitActionFlyoutItem_Click(object sender, RoutedEventArgs e)
+    {
+        if(sender is MenuFlyoutItem actionFlyoutItem)
+        {
+            (AIToolkitAction action, ModelDetails modelDetails) = ((AIToolkitAction, ModelDetails))actionFlyoutItem.Tag;
+
+            AIToolkitActionClickedEvent.Log(AIToolkitHelper.AIToolkitActionInfos[action].QueryName, modelDetails.Name);
```

Comment:
Would be good to also capture whether the deep link worked or we had to fallback to the ai toolkit docs. Can you add a property (bool) and call this below?